### PR TITLE
Add `bind_layer()`.

### PR DESCRIPTION
**Pull request by apghml**

### CODE REVIEW

File: axlearn/common/test_utils.py [hunk 1]
```diff
@@ -708,3 +713,66 @@ def temp_chdir(new_cwd: Union[pathlib.Path, str]):
...
+
+    Returns:
+        The Initialized module.
+    """
+    if prng_key is None:
+        prng_key = jax.random.PRNGKey(0)
+
+    init_key, ctx_key = jax.random.split(prng_key)
+    if isinstance(layer, InstantiableConfig):
+        if isinstance(layer, BaseLayer.Config) and isinstance(
```
[the remark this answers is not in the document]
Comment (apghml) [reply axlearn/common/test_utils.py:761]: A `Module` doesn't necessarily support `initialize_parameters_recursively()`.

File: axlearn/common/test_utils.py [hunk 2]
```diff
@@ -708,3 +713,66 @@ def temp_chdir(new_cwd: Union[pathlib.Path, str]):
         yield
     finally:
         os.chdir(old_cwd)
+
+
+L = TypeVar("L", bound=BaseLayer)
+
+
+@contextlib.contextmanager
+def bind_layer(
```
[the remark this answers is not in the document]
Comment (apghml) [reply axlearn/common/test_utils.py:722]: I considered this, but a Module doesn't necessarily support initialize_parameters_recursively().

File: axlearn/common/test_utils.py [hunk 3]
```diff
@@ -708,3 +713,66 @@ def temp_chdir(new_cwd: Union[pathlib.Path, str]):
...
+    prng_key: Optional[jax.random.PRNGKey] = None,
+    state: Optional[Nested[Tensor]] = None,
+) -> Iterator[L]:
+    """Creates a context in which `module` has state initialized using `init_method`.
+
+    This lets you write tests that make calls to a module without needing to call `functional()`
+    yourself.
+
+    It is similar in spirit to FLAX's `module.bind()` although that works differently due to the
+    fact that FLAX state is only associated with an instane of a module, whereas AXLearn state is
```
[the remark this answers is not in the document]
Comment (apghml) [reply axlearn/common/test_utils.py]: Done.

File: axlearn/common/test_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (apghml) [reply axlearn/common/test_utils.py:722]: One option would be to define a separate `bind_module()` and then implement most of `bind_layer()` via a call to `bind_module()`.

WDYT?

[the remark this answers is not in the document]
Comment (ruomingp) [reply axlearn/common/test_utils.py:722]: sgtm

**Review by ruomingp — APPROVED** [review]

**Pull request merged**